### DISCUSSION TITLE
can't delete a couple short urls

### DISCUSSION THREAD

**Discussion by Stunt0265** [shlinkio/discussions#105] [Help wanted]
I"m running the latest version of Shlink on Docker and using PostgreSQL database.  I have a couple short urls I'm unable to delete in the webui or the CLI.  I can see them and they work, but the CLI doesn't even seem to know they exist when I try to view it using  short-url:parse.  Probably database corruption based on the recent work I was doing.  Any ideas to clean this up, I can access the database with pgadmin.

**Comment by Stunt0265** [comment]
Turns out I have duplicates from an import.  So I have two of each and I can delete one, but not the remaining one.

**Comment by acelaya** [comment]
> I have a couple short urls I'm unable to delete in the webui or the CLI

What do you mean by this? Does it throw some kind of error?

> Turns out I have duplicates from an import.

Did you specify some specific domain when importing? Can you share a screenshot with the information in the database for some of those URLs that are "duplicated"?

**Comment by Stunt0265** [comment]
No error, just doesn't do anything.  Either in the web UI or CLI.  I did specify a specify a domain during import, and different than the main domain at the time.  Can you tell me what table to find what you need to share?

**Comment by Stunt0265** [comment]
I think I found the issue.  The short_urls table has missing domain_id for many entries and most are missing author_api_key_id.

**Comment by acelaya** [comment]
> I think I found the issue. The short_urls table has missing domain_id for many entries and most are missing author_api_key_id.

That's expected. Those are not mandatory fields.

* Null value on `domain_id` means it is linked to the default domain, and will also work from any domain.
* Null value on `author_api_key_id` means it was created without specifying an API key. Typically because it was created from the CLI (like when you import short URLs).

> No error, just doesn't do anything. Either in the web UI or CLI. I did specify a specify a domain during import, and different than the main domain at the time.

Shlink allows the same "short URL" (or the same short code/custom slug to be more precise) to be used on multiple domains.

This means you can have `DEFAULT_DOMAIN=foo.com`, and create a URL with custom slug `whatever`, resulting in `foo.com/whatever`.

Later, Shlink allows you to create another URL with custom slug `whatever`, only if you specifically provide a domain, let's say `bar.com`, resulting in `bar.com/whatever`.

If you try to interact with those via API or CLI, and do not specify the domain, Shlink will always fall back to the first one, which was created for the default domain (so, no domain specifically provided). This is a common mistake when interacting with Shlink's CLI.

However, shlink-web-client should know if it has to provide the domain, so there might be a bug there. No likely that the bug is in the server.

That's why I was asking to get a screenshot of some of those "duplicated" short URLs, because, specially when importing, multiple meta-values are set in internal columns.

Getting a screenshot of the domains table would also help.

Also, please, make sure you provide all the info from the [bug issue template](https://github.com/shlinkio/shlink/blob/f7ae52f86e1294fccdd7740a7046ef012b616c90/.github/ISSUE_TEMPLATE/Bug.md), as you skipped everything and that helps me reproduce potential bugs.

Ultimately, take a look at these pages, which provide a bit more clarity on some of the behaviors involved here:

* https://shlink.io/documentation/advanced/multiple-domains/
* https://shlink.io/documentation/advanced/import-short-urls/
* https://shlink.io/documentation/api-docs/api-key-roles/#considerations

**Comment by Stunt0265** [comment]
When I first imported from another Shlink instance to the new instance and with a new base domain, I had 500 urls with domain.info.  I then imported from a csv file and specified a domain of domain.link.  So ended up with 500 urls with base domain of domain.info and 500 of domain.link.  I then changed the DEFAULT_DOMAIN and SHLINK_SERVER_URL to domain.link from domain.info.  After this, I had duplicate links all using base domain of domain.link.  But 500 seemed to respond to domain.info.  And domain.link and domain.info has identical DNS records.  I'll provide screenshots shortly.

**Comment by Stunt0265** [comment]
I sent attachments to your gmail. Let me know if you prefer other direct method.

Thanks

**Reply by Stunt0265** [reply]
Let me know if you didn't receive the email with the screenshots and text files.

Thanks

**Comment by Stunt0265** [comment]
Any more ideas on how I might fix these issues?

**Comment by Stunt0265** [comment]
Circling back on this - is there any way I can delete these URLs?
